### PR TITLE
Prefer sysroot matching subdir when there are multiple

### PR DESCRIPTION
**Pull request by isuruf**
### Description

Improves sysroot selection.

Fixes https://github.com/conda/conda-build/issues/5403

### Checklist - did you ...

- [x] Add a file to the `news` directory ([using the template](../blob/main/news/TEMPLATE)) for the next release's release notes?
     
- [x] Add / update necessary tests?
- [ ] Add / update outdated documentation?

### CODE REVIEW

**Comment by h-vetinari** [comment]
I tested this PR in https://github.com/conda-forge/matplotlib-feedstock/pull/398 and it works! Great work!

File: conda_build/post.py [hunk 1]
```diff
@@ -1406,8 +1412,19 @@ def check_overlinking_impl(
                     list(diffs)[1:3],
                 )
                 sysroots_files[srs] = sysroot_files
+
+    def sysroot_matches_subdir(path):
+        if path.endswith("/"):
+            path = os.path.dirname(path)
```
Comment (kenodegard) [inline conda_build/post.py:1418]: It looks like all of the paths will end with `/`, see line 1374 above:

```python
srs = sysroot if sysroot.endswith("/") else sysroot + "/"
```

So we can simplify this and remove the conditional

File: conda_build/post.py [hunk 2]
```diff
@@ -1406,8 +1412,19 @@ def check_overlinking_impl(
                     list(diffs)[1:3],
                 )
                 sysroots_files[srs] = sysroot_files
+
+    def sysroot_matches_subdir(path):
```
Comment (kenodegard) [inline conda_build/post.py:1416]: Let's include a comment here with an example of what the `sysroot` path will look like. Make it easier to understand what we are slicing out of the pathname.

**Review by kenodegard — CHANGES_REQUESTED** [review]
@isuruf I see this is still in draft mode, if you're not ready for reviews feel free to ignore these comments

File: conda_build/post.py [hunk 3]
```diff
@@ -1406,8 +1412,20 @@ def check_overlinking_impl(
...
+        # The path looks like <PREFIX>/aarch64-conda-linux-gnu/sysroot/
+        # We check that the triplet "aarch64-conda-linux-gnu"
+        # matches the subdir for eg: linux-aarch64.
+        triplet = os.path.basename(os.path.dirname(os.path.dirname(path)))
+        subdir_arch = subdir.split("-")[-1]
+        return triplet.split("-")[0] == GNU_ARCH_MAP.get(subdir_arch, subdir_arch)
```
Comment (kenodegard) [inline conda_build/post.py:1422]:
```suggestion
        sysroot_arch = Path(path).parent.name.split("-")[0]
        subdir_arch = subdir.split("-")[-1]
        return sysroot_arch == GNU_ARCH_MAP.get(subdir_arch, subdir_arch)
```

**Review by kenodegard — APPROVED** [review]
Minor suggestion, otherwise LGTM!

**Review by kenodegard — APPROVED** [review]

**Review by beeankha — APPROVED** [review]

**Comment by kenodegard** [comment]
`tests/test_api_build.py::test_sysroots_detection` implemented here is now failing on `main`, see https://github.com/conda/conda-build/actions/runs/10671805383/job/29578195906

**Pull request merged**